c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -196,9 +236,14 @@ class DeviceCachingAllocator {
   // unallocated cached blocks 1 MB or smaller
   BlockPool small_blocks;
 
-  // allocated or in use by a stream
+  // allocated or in use by a stream. Holds all active allocations,
+  // whether they came from graph_pools or one of the BlockPools above.
   std::unordered_set<Block*> active_blocks;
 
+  // captures_underway tracks if a capture might be underway on any stream.
+  // Most of the time it's zero, in which case malloc can avoid calling
+  // cudaStreamGetCaptureInfo in the hot path.
+  int captures_underway = 0;
```

Comment:
There's a stronger invariant here, something like captures underway equals the sum of all use counts of pools. If we had some slow invariant checker for the caching allocator that would be a good one to check.

Reply:
> captures underway equals the sum of all use counts of pools

I don't think this is an invariant, because captures_underway is only nonzero during capture. Post capture, each graph object (and corresponding use_count increment) lasts much longer. Or am i misinterpreting?